### ISSUE TITLE
Compare Exchange Hashes - Exchange 2013 - js files seem to be minified - should they also be excluded?

### ISSUE THREAD

**Issue by jaminbern** [microsoft/CSS-Exchange#282]
Hi - I notice that you have this line of code to ignore specific js files:
($f.Name.EndsWith(".strings.localized.js") -eq $false)

All of the files I'm getting marked as suspicious end with "Strings.localized.min.js"

They are also in an old exchange Directory: 15.0.847.32

I don't want to assume these can be ignored (I don't really know why the others can be). 

Can you please check and confirm (and update your script if it IS ok to ignore them?)

Thanks

Jaminbern

**Comment by MvanGroenigen** [comment]
Version 15.0.847.32 was added recently

https://github.com/microsoft/CSS-Exchange/blob/main/Security/src/Baselines/baseline_15.0.847.32.csv 

You may need to re-download the script and rerun.

**Comment by jaminbern** [comment]
I copied the source from git at about 09:50 GMT  today (an hour 30 mins ago)- none of the baselines have been changed in the last 6 hours ?

**Comment by N1c093** [comment]
I also run the `CompareExchangeHashes.ps1` one hour ago on Exchange 2013. As @jaminbern already mentioned every `...strings.localized.min.js` file gets marked as suspicious.

**Comment by dstjean** [comment]
@jaminbern I've provided additional info in ticket #300

**Comment by jaminbern** [comment]
Thanks dstjean. I note we have around 1480 or so of these - so less than you - but that might be to do with localization. We are showing about 25 different locale's...

**Comment by psykokwak4** [comment]
could you share with us sample results?

**Comment by jaminbern** [comment]
[NoHashMatch.txt](https://github.com/microsoft/CSS-Exchange/files/6128481/NoHashMatch.txt)

**Comment by jaminbern** [comment]
psykokwak4 - could you please update your profile so we know who you are?

**Comment by psykokwak4** [comment]
from your example results:
```
"C:\Program Files\Microsoft\Exchange Server\V15\ClientAccess\owa\prem\15.0.847.32\
"C:\Program Files\Microsoft\Exchange Server\V15\ClientAccess\owa\prem\15.0.847.45\
```

so script finds server's previously installed exchange versions then try to fetch them locally or from github.
but previously it only finds installed versions under top sub directory of VDIR physical paths 
so for example for OWA VDIR:
 `VDIR "Exchange Back End/owa/" (physicalPath:C:\Program Files\Microsoft\Exchange Server\V15\ClientAccess\owa)`
script was only able to find versions like
`C:\Program Files\Microsoft\Exchange Server\V15\ClientAccess\owa\15.2.792.5`
but not versions like
`C:\Program Files\Microsoft\Exchange Server\V15\ClientAccess\owa\prem\15.2.792.10`
this issue was fixed 2 days ago(commit 74165688bac6fd29d852bc3bc1d08b67c98b7e53), if you use the latest script from (https://github.com/microsoft/CSS-Exchange/releases/download/v21.03.12.0547/CompareExchangeHashes.ps1), it should be able to return you 15.0.847.32

then you will hit the next issue, that 15.0.847.45 files will be marked as NoHashMatch since we don't have it in `$VALID_VERSIONS ` dictionary([L130](https://github.com/microsoft/CSS-Exchange/blob/fb3b05c86d390a40d1b8d91eb47546f5e400dd44/Security/src/CompareExchangeHashes.ps1#L130)), and have not yet prepared its baseline

**Comment by psykokwak4** [comment]
also, RE on "Strings.localized.min.js" errors, we don't have them in baseline (not that we don't have those files)
closing this issue, and if baselines will be updated with *Strings.localized.min.js" hashes, this issue should be fix.
please check issue `#313` for baseline release
and feel free to reopen this issue
